### DISCUSSION TITLE
Question: how do I update latest cwl updates?

### DISCUSSION THREAD

**Discussion by foreachthing** [texstudio-org/texstudio#3729] [Q&A] [answered]
I've seen so many updates to cwl files and I was wondering how to reflect these updates on my computer? What is the best practice?

**Comment by muzimuzhi** [comment, marked as answer]
If you don't mind using developing versions of TeXstudio, then you can download artifact from [latest successful run of `cd.yml`](https://github.com/texstudio-org/texstudio/actions/workflows/cd.yml). For example artifacts of [run #9869550136](https://github.com/texstudio-org/texstudio/actions/runs/9869550136).

**Reply by foreachthing** [reply]
Hmm ... and I thought there was an easy way :-)
Well, for now I'll just clone/pull to a directory and copy the cwl files manually. Thanks though!

**Comment by octaeder** [comment]
I think copying is not sufficient. Looking into files of PR #3728 shows a change of `completion.qrc` related to cwls. So I assume a build is necessary. Nevertheless, you may generally not experience any problems after copying the cwls, except for the cwls that have been added to the `completion.qrc` file.

**Reply by muzimuzhi** [reply]
As far as cwl syntax is not extended, copying to [user cwl directory](https://github.com/texstudio-org/texstudio/wiki/Frequently-Asked-Questions#where-are-cwl-files-stored) should work, with some efficiency loss.
